model.py

```diff
@@ -3,45 +3,34 @@
 class Model:
     data_access = Dao()
 
-    def __init__(self, name, function, _id):
-        self.name = name
-        self.function = function
-        self._id = _id
+    def __init__(self, params):
+        self.name = params['name']
+        self.function = params['function']
+        self._id = str(params['_id'])
+
     @classmethod
-    def update(cls, _id, name, function):
-        updates = {}
-        if name:
-            updates['name'] = name
-        if function:
-            updates['function'] = function
-        response = cls.data_access.update(_id, updates)
+    def update(cls, _id, params):
+        response = cls.data_access.update(_id, params)
         print(response)
         return response
 
     @classmethod
     def save(cls, payload):
-       cls.data_access.add(payload)
-       return cls.from_dict(payload)
+       _id = cls.data_access.add(payload)
+       return {'_id' : _id}
```

Comment:
return the payload with the new id added to it .. also the model will always return instances or an instance of itself .. 

```py
payload['id'] = _id
return Model(payload)
```